heareval/predictions/task_predictions.py

```diff
@@ -253,6 +254,40 @@ def validation_step(self, batch, batch_idx):
     def test_step(self, batch, batch_idx):
         return self._step(batch, batch_idx)
 
+    def log_scores(self, name: str, score_args):
+        """Logs the metric score value for each score defined for the model"""
+        assert hasattr(self, "scores"), "Scores for the model should be defined"
+        end_scores = {}
+        for score in self.scores:
+            score_ret = score(*score_args)
+            validate_score_return_type(score_ret)
+            # If the returned score is a tuple, store each subscore as separate entry
+            if isinstance(score_ret, tuple):
+                # The first score in the returned tuple will be used
+                # as primary score for this metric
+                end_scores[f"{name}_{score}"] = score_ret[0][1]
+                # All other scores will also be logged
+                for (subscore, value) in score_ret:
+                    end_scores[f"{name}_{score}_{subscore}"] = value
+            elif isinstance(score_ret, (float, int)):
+                end_scores[f"{name}_{score}"] = score_ret
+            else:
+                raise ValueError(
+                    f"Return type {type(score_ret)} is unexpected. Return type of "
+                    "the score function should either be a "
+                    "tuple(tuple(str, (float, int))) or float or int. "
+                )
+
+        # Weird, scores can be nan if precision has zero guesses
+        for score_name, value in end_scores.items():
+            end_scores[score_name] = 0.0 if math.isnan(value) else value
```

Comment:
Wait. No. Why would we do this? NaN is NaN. It's wrong to convert it to zero. Particularly for a score we want to minimize not maximize. This is bad.

Reply:
removing this ( was there from before )